### ISSUE TITLE
Use of AWE32 filter cutoff NRPN overrides both cutoff and emphasis

### ISSUE THREAD

**Issue by mrbumpy409** [FluidSynth/fluidsynth#1452]
### FluidSynth version
2.4.1 (1427 branch)

### Describe the bug
FluidSynth now supports AWE32 NRPN control of both filter cutoff (FC) and emphasis (Q). The use of the FC NRPN should override the preset's FC, and the use of the Q NRPN should override the preset's Q. However, using the FC NRPN overrides both FC and Q (which ends up being set to 0, as far as I can tell).

### Expected behavior
Using the FC NRPN should only override the preset's FC and not the Q.

### Steps to reproduce
This bug can be reproduced pretty easily by running the test linked in [this comment](https://github.com/FluidSynth/fluidsynth/pull/1434#issuecomment-2555507683). A spectrogram view of tests 1 and 3 both show the presets' Q values being overridden, as these two tests make use of the FC NRPN.

**Comment by derselbst** [comment]
Right, the reason I was implementing it this way, is because of how the "NRPN LSB 22 (Initial Filter Resonance Coefficient)" is documented:

* [SB AWE32 Developer's Information Pack](https://github.com/user-attachments/files/15757220/adip301.pdf)
* [Frequently Asked Questions for SB AWE32](http://archive.gamedev.net/archive/reference/articles/article445.html)

There, they specify a coefficient table with a high and low Q, and a high and low filter fc respectively. Therefore, I assumed that the NRPN implementation of the filter uses a varying filter Q, i.e. when setting the filter fc to its lower bound, you'll get the low Q. If you set fc to its upper bound, you'll get a high filter Q. And any fc in between just linearly interpolates the Q within its given range.

I didn't had time so far to look into your provided test. But I'll probably find time to do so over Christmas. In addition, you may try to add another test case to it which sets the NRPN LSB 22 (i.e. initializes the Filter Resonance Coefficient), and then vary the filter fc via NRPN LSB 21. I would assume that this also has an effect on the filter's emphasis.

**Comment by mrbumpy409** [comment]
At least on the Audigy, the filter when modulated by NRPN LSB 21 acts the same as if it were modulated by any other source (via modulator, etc.), with no affect on the filter Q. You can see this in the test results on the other thread. With FluidSynth aiming to be a SoundFont 2.01/2.04-compliant synthesizer, the Audigy1/2 behavior would be the ideal model to emulate, IMO. When I get some time, I'll update the test to include filter Q tests.

I hope you have a Merry Christmas! I am very thankful for you and all of the work you put into this synthesizer! May you be safe and warm through the holidays. :grin:

**Comment by mrbumpy409** [comment]
I have made a new version of the filter NRPN tests along with a separate bug report covering the discovered issues here: #1473

**Comment by derselbst** [comment]
#1516 will address this by no longer handling NRPN filter Q dependent on NRPN filter Fc.

**Comment by klerg** [comment]
Nice to know and glad you were able to fix this fairly soon even though it was not so easy to do this
